### PR TITLE
Update sorting.d

### PR DESCRIPTION
**Pull request by bachmeil**
Added link to Rosetta Code example.

### CODE REVIEW

File: std/algorithm/sorting.d [hunk 1]
```diff
@@ -1949,6 +1949,8 @@ sort!((a, b) => hashFun(a) < hashFun(b))(array);
 schwartzSort!(hashFun, "a < b")(array);
 ----
 
+See also $(WEB http://rosettacode.org/wiki/Sort_an_array_of_composite_structures#D, this example) on Rosetta Code.
+
```
Comment (JackStouffer) [inline std/algorithm/sorting.d]: This should be reformatted to be
```
See Also:
$(WEB ...)
```
So it's is properly recognized and formatted.

**Comment by schuetzm** [comment]
Hmm... `schwartzSort` looks nice for this task, but is inefficient, because it allocates an array to cache the sort keys. This is of course overkill in our case.

@nordlow wanted to get a `sortBy` into Phobos:
http://forum.dlang.org/thread/nqwzojnlidlsmpunpqqy@forum.dlang.org
Has that been submitted yet? If we have `sortBy`, then the RosettaCode example should probably be changed to use that, and it shouldn't be linked from `schwartzSort`.

**Comment by bachmeil** [comment]
Okay, I saw that Rosetta code example and thought it was clearer than the current example.

Would the following be accurate:

> `schwartzSort` returns the same output as an equivalent call to `sort` but it is faster when the sort comparison requires an expensive comparison. `schwartzSort` allocates a temporary array to cache the sort keys, so it can be slower than `sort` in other cases.

I think it's best if we precisely state right away why you would and wouldn't want to use `schwartzSort` rather than a regular sort. This is the first I've heard of this algorithm and the current documentation assumes the reader has used it in another language.

**Comment by bachmeil** [comment]
Closing as that is not a proper example for `schwartzSort`.

**Pull request closed without merging**